distributed/tests/test_client.py

```diff
@@ -7571,3 +7572,21 @@ def __init__(self, *args, **kwargs):
         (DeprecationWarning, "The io_loop property is deprecated"),
         (DeprecationWarning, "setting the loop property is deprecated"),
     ]
+
+
+@gen_cluster(client=False, nthreads=[])
+async def test_fast_close_on_aexit_failure(s):
+    class MyException(Exception):
+        pass
+
+    c = Client(s.address, asynchronous=True)
+    with mock.patch.object(c, "_close", wraps=c._close) as _close_proxy:
+        with pytest.raises(MyException):
+            async with c:
+                start = time()
+                raise MyException
+        stop = time()
+
+    assert _close_proxy.mock_calls == [mock.call(fast=True)]
+    assert c.status == "closed"
+    assert (stop - start) < 2
```

Comment:
I couldn't find any tests that tested what `_close(fast=True)` does:

it just reduces this 2 second timeout to 0
https://github.com/dask/distributed/blob/c373746a1fae0803be484474042bfdc18ab01699/distributed/client.py#L1584

Reply:
introduced here https://github.com/dask/distributed/pull/27/files#diff-ce261e4d4b0185c8defbfcd46ebe8311a3ecb7ab81d653a2c6664c840340ab2cR250-R251

Reply:
which is this: https://trio.readthedocs.io/en/stable/reference-io.html#trio.aclose_forcefully

Reply:
@hendrikmakait thanks for the review, it helped clarify my thinking on this!

Reply:
nit: Would it make sense to rename `fast` to `force`?

Reply:
Yeah I think "forcefully=True" is probably the best here